ctp2_code/ui/aui_common/aui_pixel.cpp

```diff
@@ -45,8 +45,8 @@ uint8 aui_Pixel::GetPaletteIndexedColor(uint8 red, uint8 green, uint8 blue, HPAL
 	uint16          valMain         = ColorCode555(red, green, blue);
 	uint8           color           = 0;
 	sint32          diff            = INT_MAX;
-
-	for (uint8 i = 0; i < static_cast<uint8>(paletteCount); ++i)
+	// palleteCount must be 256
+	for (uint8 i = 0; i <= 255; ++i)
```

Comment:
This piece still smells. In principle, `i` could still become 255 and then be incremented which would lead to an overrun and `i` becomes 0. So either turn `i` into a uint16 or sint16, or add a break line at the end of the loop like `if(i == 255) break;`.

Reply:
You're right - I was assuming this routine would always find a match, and therefore that overflow would never happen (as otherwise these routines would never have worked).

I have now searched through the history of these routines and they have never actually been used in all of the git history (though they have been refactored) - this is why the fact they were incorrect didn't seem to break anything. Should I just do a patch removing them and the supporting routine entirely?